### PR TITLE
WIP/Feedback: Homepage welcome message (#740)

### LINKED ISSUE #740: More descriptive home page
Whenever Invidious gets linked to, people are left confused what it actually is ([latest example](https://old.reddit.com/r/firefox/comments/czy3i5/invidition_youtube_to_invidious_redirector_addon/ez4dw88/)).

I propose adding some informational text at the top of the home page for logged-out users, that explains what Invidious is, some of its main features and provides a call to action (register and import subscriptions). Final text tbd.

Secondly, I'd also switch to showing the *popular* page by default, as it is a better representation of what the instance's registered users actually watch. *Popular* also hosts many familiar channels, letting users know that the content is actually from Youtube, and not hosted by Invidious itself. (videos on the _top_ page aren't representative, as they can appear after only 1 person watched them through Invidious and often are in foreign languages and/or from not widely known personalities)

I've hacked up a quick mock-up in the devtools, but if this is something that is accepted, I can create a proper pull request and put some more effort into a nice looking UI and making it e.g. dismissable.

screenshot

<details><summary>code of the mockup (click to expand)</summary>

```html
<!-- inserted instead of the popular/top/trending bar -->
<div>
    <h1>Welcome to Invidious</h1>
    <p>Invidious is a featureful Youtube-player, that respects your privacy.
        </p><p>Features:
        </p><ul>
            <li>Audio-only mode
</li><li>Dark mode
</li><li>No ads
</li><li>Lightweight
</li><li>Doesn't require a Google account
</li><li>Support for Reddit comments
</li><li> ... and others! Click to <a href="https://github.com/omarroth/invidious/blob/master/README.md">learn more</a>
    </li></ul>
        <p><button>create an account and import your subscriptions now</button>
    </p><p>Here is a selection of videos that our community is interested in:
    </p><div class="pure-u-1 pure-u-md-1-4"></div>
</div>
```

</details>

### PR DESCRIPTION
**Pull request by girst**
# rationale

New users are often confused what invidious is. Adding a welcome message (similar to e.g. GitHub's or old.Reddit's) gives us the opportunity to introduce ourselves and provide the user with a 'call to action' (signing up). 
The wording is kept mostly jargon-free (e.g. "youtube-player" instead of "-frontend") to be friendly to non-technical folks. 

~~Another change is more subtle: some users think, invidious is an alternative video hosting site. Switching the home page to 'popular' by default shows the user a bunch of familiar channels/faces they might recognize, emphasizing that content comes from Youtube.~~ *already implemented elsewhere now*

The data import page has been updated as well, with a simple two-step guide for the most common action, importing subscriptions from Youtube.

# requested feedback

 - We need to decide on which roughly 5 features we want to advertise on the homepage and the final wording of the welcome message and call to action.

Text right now:
> **Welcome to Invidious**
>   Watch Youtube-videos *Ad-free*, *download videos, audio and subitles*, *subscribe to channels* without Google tracking you and [much, much more](/).
> [Click to learn more or create an account to import your subscriptions](/) now!

 - also, is using the preferences a good place to keep track of whether the user has dismissed the message?

# TODO

- after the text has been finalized, it needs to be localized (how?). 
- rebase/squash patch

----

screenshot welcome message

fixes #740

### CODE REVIEW

**Comment by girst** [comment]
I'm stuck; maybe a crystal-expert can help me out here: I've generalized the dismissal system, so more than one infobox could be dismissed in the future. For this, I wanted to store which ones have been dismissed in either an Array(String) or a Hash(String, Bool) and push an element for each infobox. But then the preference appears to be shared between users!? If I dismiss it in one session, it's gone for all others too!

**Comment by omarroth** [comment]
Thanks for this! Apologies for not jumping on it sooner.

As mentioned in [#227](https://github.com/omarroth/invidious/issues/227#issuecomment-466800711) and #457, I would very much like to have this as an `/about` page, so it doesn't take up as much space on the homepage. Having a dismissal system works, however I think having a separate page is quite a bit simpler and provides more room for any additional information (FAQ, etc). 

Language is good. Feature selection looks good.

I agree "Popular" makes sense as the default home page. I expect the default to be changed as part of #295 so that it's still possible for current users to switch back if they prefer.

**Comment by girst** [comment]
On Fri, Oct 11, 2019 at 11:50:35AM -0700, Omar Roth wrote:
>Thanks for this! Apologies for not jumping on it sooner.

no worries, we're all just doing this in our spare time!

> I would very much like to have this as an `/about` page, so 
>it doesn't take up as much space on the homepage. Having a dismissal 

an about page would be nice to have too, but I think, showing a message 
on first arrival is important. I can tune it to take up less vertical 
space though (I'll put together a rough mock-up).

reason being that most people won't click a link to the about page, but 
just leave the homepage confused.

>system works, however I think having a separate page is quite a bit 
>simpler and provides more room for any additional information (FAQ, 
>etc).

the "and many more" link could link to that /about page (possibly with a 
different label). again: i think it is very important that we give 
_some_ message on the homepage, even if we trim it down considerably.

>
>Language is good. Feature selection looks good.
>
>I agree "Popular" makes sense as the default home page. I expect the 
>default to be changed as part of #295 so that it's still possible for 
>current users to switch back if they prefer.

the way I've implemented the current version of the patch allows this 
already, no? I can submit that as its own PR if you prefer.

**Comment by girst** [comment]
mockup
<details>
```
<div style="border-bottom: 1px solid #ddd;">
  <a style="float:right" href="/dismiss_info?name=welcome">dismiss</a><h1 style="font-size:1.4em">Welcome to Invidious</h1>

<p>Invidious is a community developed Youtube-player that respects your privacy. Features include
  </p><ul>
    <li>Ad-Free
      </li><li>Audio-only mode
        </li><li>Subscriptions without a Google account
            </li><li>Download videos, audio and subtitles
  </li><li><a href="/about">Click to learn more</a>
    </li></ul>


</div>

```
</details>

mockup
<details>
```

<div style="border-bottom: 1px solid #ddd;border-top: 1px solid #ddd;">
  <a style="float:right;margin: .67em;" href="/dismiss_info?name=welcome">dismiss</a><h1 style="font-size:1.4em">Welcome to Invidious</h1>

<p>Invidious is a community developed Youtube-player that respects your privacy. It is <b>Ad-Free</b>, has an <b>Audio-only mode</b>, lets you have <b>Subscriptions without a Google account</b>, <b>Download videos, audio and subtitles</b> and much, much more!
  </p><p><a href="/about">Click to learn more</a>
</p></div>


```
</details>

**Comment by girst** [comment]
i have made the welcome message much smaller and reworded it (only 3 features instead of 5). screenshot and text in pullreq-description.

File: src/invidious/views/components/welcome_message.ecr [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<div class="welcome-outer" id="welcome-outer">
+    <div class="infobox welcome-box">
+        <a id="dismiss_welcome" class="pure-button" href="/dismiss_info?name=welcome&amp;referer=<%= env.get?("current_page") %>">dismiss</a>
+        <h1>Welcome to Invidious</h1>
+        <p>Watch Youtube-videos <em>Ad-free</em>, <em>download videos, audio and subitles</em>, <em>subscribe to channels</em> without Google tracking you and <a href="https://github.com/omarroth/invidious/blob/master/README.md" id="learn_more_link">much, much more</a>.
```
Comment (comradekingu) [inline src/invidious/views/components/welcome_message.ecr:5]:
```suggestion
        <p>Watch YouTube-videos <em>ad-free</em>, <em>download videos, audio and subtitles</em>, <em>subscribe to channels</em> without Google tracking you and <a href="https://github.com/omarroth/invidious/blob/master/README.md" id="learn_more_link">much, much more</a>.
```

File: locales/en-US.json [hunk 2]
```diff
@@ -38,6 +38,9 @@
     "Import FreeTube subscriptions (.db)": "Import FreeTube subscriptions (.db)",
     "Import NewPipe subscriptions (.json)": "Import NewPipe subscriptions (.json)",
     "Import NewPipe data (.zip)": "Import NewPipe data (.zip)",
+    "To import your subscriptions from YouTube,": "To import your subscriptions from YouTube,",
+    "click here to download your YouTube-Takeout": "click here to download your YouTube-Takeout",
```
Comment (comradekingu) [inline locales/en-US.json:42]:
```suggestion
    "click here to download your YouTube-Takeout": "click here to download your YouTube Takeout",
```
As per https://en.wikipedia.org/wiki/Google_Takeout

File: src/invidious/views/data_control.ecr [hunk 3]
```diff
@@ -3,22 +3,26 @@
 <% end %>
 
 <div class="h-box">
+    <div class="infobox">
+        <%= translate(locale, "To import your subscriptions from YouTube,") %>
+        <a rel="noopener" target="_blank" href="https://www.youtube.com/subscription_manager?action_takeout=1">
+            <%= translate(locale, "click here to download your YouTube-Takeout") %>
```
Comment (comradekingu) [inline src/invidious/views/data_control.ecr:9]:
```suggestion
            <%= translate(locale, "click here to download your YouTube Takeout") %>
```

**Comment by girst** [comment]
Those changes LGTM, thanks! Are you interested in working on it? I don't have the time (and since built my own invidious-alternative, subscriptions.gir.st) to fix the (trivial) merge conflicts, so feel free to push more commits as you please.

**Comment by comradekingu** [comment]
@girst
"Click here to import your subscriptions from YouTube in the form of a \"YouTube Takeout\" file you can send in below."

Oh, I understood what I was meant to do now ;)

> Welcome
> This is one of many Invidious instances; where you can watch and download video, audio and subtitles ad-free. Free [your YouTube subscriptions] from Google by [making an account here]. Avoid even the last bit of tracking by using the [Tor Browser].

**Review by comradekingu — CHANGES_REQUESTED** [review]
Commented in the thread :)

**Comment by girst** [comment]
On Sun, Sep 13, 2020 at 01:42:29PM -0700, Allan Nordhøy wrote:
>@girst
>"Click here to import your subscriptions from YouTube in the form of a \"YouTube Takeout\" file you can send in below."
>
>Oh, I understood what I was meant to do now ;)

not sure what you want from me here. 

>> Welcome
>> This is one of many Invidious instances; where you can watch and download video, audio and subtitles ad-free. Free [your YouTube subscriptions] from Google by [making an account here]. Avoid even the last bit of tracking by using the [Tor Browser].

i guess this is a requested update to the welcome message? as i said 
earlier, please make any changes to this pr yourself (i believe you have 
commit access?); i don't have the time to work on invidious any more.

however, i would suggest against recommending the tor browser. video 
streaming is really hard on the network, and i see no indication that 
google is using googlevideo-streams to track people (it afair doesn't 
even use cookies or etags).

**Comment by SamantazFox** [comment]
bump, still relevant (though it probably will require to be redone from zero, as code changed a lot).

**Comment by syeopite** [comment]
Sure, it may still be relevant but the PR is essentially abandoned; no one has forked and continued working on it either. It may be better to just close the PR*. Either way, seeing as this is still an issue I'm just going to go ahead and open #740 again.

*This is mostly just my take on the stale PR handler and why I shortened the length. PRs should be *active* imo and anything that hasn't received any activity for more than 45 days can likely be considered abandoned.

**Comment by SamantazFox** [comment]
yeah, but the idea's good, and there was already some work / review done, so that would be sad to dump it. The welcome message is simple and looks good, though there are some improvements to make (for translations, especially).

**Pull request closed without merging**